docs/Custom_Device_Tips_and_Tricks.md

```diff
@@ -0,0 +1,271 @@
+•	Unload SDF?—Indicates whether or not the system definition file was unloaded. Unload SDF? is always True.<br />
+•	Saved?—Indicates whether or not a user saved the system definition file before closing System Explorer.<br />
+•	Path—Path on disk to the system definition file.<br />
+•	System Explorer Shutdown?—Indicates whether or not System Explorer closed. This parameter is always True.<br />
+
+**ActionVIOnSave**
+Executes when a user saves the system definition file. For example, you can customize this template to log each time the custom device is saved.
+
+**ActionVIOnDownload**	
+Executes when a user deploys the system definition file containing the custom device to a real-time target. This action VI does not execute if a user deploys the system definition to a Windows target.
+
+This template helps create action VIs that finalize the target configuration after you deploy the system definition. You can also customize this template to deploy any additional files or dependencies your custom device requires. For example, if your custom device reads and writes to shared variables, you can deploy those variables.
+
+The template has the following unique parameters.<br />
+
+•	Device Item Ref in—Reference to the custom device item whose XML declaration calls this action VI.<br />
+•	ftp session—Open FTP session used to download the system definition to the target. You can use this open session to move additional files to the target.<br />
+•	System Definition Dir—Path to the system definition file on disk.<br />
+•	IP Address—IP address of the target.<br />
+•	ftp session out—Open FTP session used to download the system definition file to the target.<br />
+
+**ActionVIOnPaste**
+Executes when a user pastes a custom device item. This template helps create action VIs that check channel properties. For example, if the user pastes a page that configures a target, you can create an action VI to ensure that the new page does not attempt to reconfigure the target.
+
+You can also customize this template to prompt a user to enter new values for the pasted item. For example, if a user pastes a page that will conflict with existing pages, you can prompt the user to enter new values for the page.
+
+The template has the following unique parameters.<br />
+
+•	Ptr in—Reference to the custom device item whose XML declaration calls this action VI.<br />
+•	Parent—Reference to the parent of the custom device item whose XML declaration calls this action VI.<br />
+•	All Ptrs—Array of references to the items the user pasted. You can only select one item to copy. This array only contains one reference that matches the Ptr in reference.<br />
+
+**ActionVIOnCompile**
+Executes when VeriStand compiles the system definition file.
+If you deploy the system definition, then undeploy it, and then redeploy it without making changes, this template does not execute because the system definition does not recompile.
```

Comment:
```suggestion

If you deploy the system definition, then undeploy it, and then redeploy it without making changes, this template does not execute because the system definition does not recompile.
```